### ISSUE TITLE
Clean IO classes to all accept consistent arguments

### ISSUE THREAD

**Issue by jgostick** [PMEAL/OpenPNM#2275]
And perhaps touch up the doc strings as well.

**Comment by jgostick** [comment]
Specifically, all classes should take network only, and not project

**Comment by jgostick** [comment]
Also, the ``phases`` argument should be more general to also accept ``algorithms``, that way you don't have to write the values from the algorithm back onto the phases.  This should also accept 'solution' objects maybe?

**Comment by jgostick** [comment]
After some discussion it seems that we probably should keep things relatively the same, meaning the user should add the ``TransientSolution`` to the phase, then pass the phase to the io as per the old way.  The IO classes need to be updated so they detect the presence of a ``TransientSolution`` and break it up into individual columns like ``pore.concentration@10`` or individual files for paraview.  

HOWEVER, it might still be useful to allow passing of algoirthm objects to the IO classes, just for flexibility.

**Comment by jgostick** [comment]
I have just culled the io classes to make them work with the "new" PostGeoPhys V3.  Here is a list of changes I made:

* Removed MAT (matlab) from IO.  This was a legacy format that other collaborators used and is not worth supporting any more.  Getting values in/out of matlab can be done via CSV if really neeed.
* Removed the passing of empty projects or networks to any of the classes.  These are created when needed inside each function.
* Removed from_vtk because importing data from this format is not a sensible thing to do
* Removed check for whether transient data is present...this needs to be figured out still (see issue #2068)
